exchanges/subscription/list.go

```diff
@@ -30,3 +59,186 @@ func (l List) GroupPairs() (n List) {
 	}
 	return s.List()
 }
+
+// QualifiedChannels returns a sorted list of all the Qualified Channels in the list
+func (l List) QualifiedChannels() []string {
+	c := make([]string, len(l))
+	for i := range l {
+		c[i] = l[i].QualifiedChannel
+	}
+	slices.Sort(c)
+	return c
+}
+
+// SetStates sets the state for all the subs in a list
+// Errors are collected for any subscriptions already in the state
+// On error all changes are reverted
+func (l List) SetStates(state State) error {
+	var err error
+	for _, sub := range l {
+		err = common.AppendError(err, sub.SetState(state))
+	}
+	return err
+}
+
+func fillAssetPairs(ap assetPairs, a asset.Item, e iExchange) error {
```

Comment:
suggestion: This would be cool as a currency pairs manager method, annoying with the formatting though.

Reply:
Noted 📓 
I'm going to smesh the formatting soon, khabib style.